### PR TITLE
BDOG-1350: Adds Alert Config

### PR DESCRIPTION
**Pull request by Tyrpix**

### CODE REVIEW

File: app/uk/gov/hmrc/serviceconfigs/scheduler/AlertConfigScheduler.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                                     ) (implicit
+                                        actorSystem         : ActorSystem,
+                                        applicationLifecycle: ApplicationLifecycle,
+                                        ec                  : ExecutionContext
+) extends SchedulerUtils {
+  
+  scheduleWithLock(
+    label = "alertConfigReloader",
+    schedulerConfig = schedulerConfigs.alertConfigUpdate,
+    lock = LockService(mongoLockRepository, "alert-config-scheduler", 20.minutes)) {
```
Comment (IanMcShane) [inline app/uk/gov/hmrc/serviceconfigs/scheduler/AlertConfigScheduler.scala:41]: could we have the time (20 minutes) configurable?

File: app/uk/gov/hmrc/serviceconfigs/service/AlertConfigSchedulerService.scala [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                                            artifactoryConnector: ArtifactoryConnector)(implicit val ec : ExecutionContext) {
+
+  private val logger = Logger(this.getClass)
+
+  def updateConfigs(): Future[Unit] = {
+
+    logger.info("Starting")
+
+    (for {
+      latestHashString  <- artifactoryConnector.getLatestHash().map(x => x.getOrElse("")) //123
```
Comment (IanMcShane) [inline app/uk/gov/hmrc/serviceconfigs/service/AlertConfigSchedulerService.scala:47]: remove the extra comments

File: project/AppDependencies.scala [hunk 3]
```diff
@@ -25,11 +25,13 @@ object AppDependencies {
...
-    "org.scalatest"     %% "scalatest"               % "3.2.3"             % Test,
-    "com.typesafe.play" %% "play-test"               % PlayVersion.current % Test,
-    "org.mockito"       %  "mockito-core"            % "3.7.7"             % Test,
-    "org.mockito"       %% "mockito-scala-scalatest" % "1.16.23"           % Test,
-    "com.typesafe.akka" %% "akka-testkit"            % "2.6.10"            % Test,
-    "uk.gov.hmrc.mongo" %% "hmrc-mongo-test-play-28" % hmrcMongoVersion    % Test
+    "org.scalatest"               %% "scalatest"               % "3.2.3"              % Test,
+    "com.typesafe.play"           %% "play-test"               % PlayVersion.current  % Test,
+    "org.scalatestplus.play"      %% "scalatestplus-play"      % "4.0.0"              % Test,
+    "com.github.tomakehurst" %  "wiremock"                    % "1.58"                % Test,
```
Comment (IanMcShane) [inline project/AppDependencies.scala:31]: alignment of text

File: test/uk/gov/hmrc/serviceconfigs/AlertConfigIntegrationSpec.scala [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+
+import scala.concurrent.Future
+
+class AlertConfigIntegrationSpec
+  extends AnyWordSpec
+    with DefaultPlayMongoRepositorySupport[AlertEnvironmentHandler]
+    with GuiceOneServerPerSuite
+    //with WireMockEndpoints
```
Comment (IanMcShane) [inline test/uk/gov/hmrc/serviceconfigs/AlertConfigIntegrationSpec.scala:38]: remove comments.

**Review by IanMcShane — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/serviceconfigs/scheduler/AlertConfigScheduler.scala (hunk 1 above)
Comment (Tyrpix) [reply app/uk/gov/hmrc/serviceconfigs/scheduler/AlertConfigScheduler.scala:41]: would this be done through config?

**Review by IanMcShane — APPROVED** [review]

**Pull request merged**